packages/ra-ui-materialui/src/list/BulkActionsToolbar.tsx

```diff
@@ -52,6 +62,13 @@ export const BulkActionsToolbar = (props: BulkActionsToolbarProps) => {
                             smart_count: selectedIds.length,
                         })}
                     </Typography>
+                    {displaySelectAllButton && (
+                        <Button
+                            label={translate('ra.action.select_all')}
+                            onClick={handleSelectAll}
+                            sx={{ ml: 1 }}
```

Comment:
We should probably have a `gap` in `BulkActionsToolbarClasses.title` instead (and remove some margins from `BulkActionsToolbarClasses.icon` too)